covid19-dashboard/dc-covid19/src/Panel.tsx

```diff
@@ -13,51 +13,67 @@
  See the License for the specific language governing permissions and
  limitations under the License.
  */
+
 import React from "react";
+import {calculate} from "./DataCalculator";
+import ContentFile from "./ContentFile.json";
 import EmptyPanel from "./EmptyPanel";
-import {difference} from "./Calculations";
-import PanelInfo from "./PanelInfo.json";
-import BarGraph from "./BarGraph";
+import {getRangeOfDates} from "./Utils";
+import Graph from "./Graph";
+import generateGraphMetadata from "./MetaDataGenerator";
 
 type Props = {
-    typeOfGraph: 'bar' | 'line',
-    data: any[],
+    allData: any[],
     label: string,
     region: string,
     datePicked: string,
     selectedShowTopN: number,
     panelId: string
 }
+type DateToGeoIdToValue = {date: {geoId: number}} | {}
+
+export default function Panel(props: Props) {
+    const content = ContentFile[props.panelId]
+    const deltaDays: number = content.deltaDays
+
+    // Is it cases or deaths?
+    const inputData: DateToGeoIdToValue = props.label === 'cases' ? props.allData[0] : props.allData[1]
+    const geoIdToPopulation: {geoId: number} = props.allData[2]
+    const geoIdToName: {geoId: string} = props.allData[3]
+    const rangeOfDates: [string, string] = getRangeOfDates(props.datePicked, deltaDays)
+
+    // Calculate per-capita.
+    const calculatedData: DateToGeoIdToValue = calculate(inputData, rangeOfDates, deltaDays, content.dataType, geoIdToPopulation)
+
+    let absoluteIncrease: DateToGeoIdToValue = {}
+    if (content.dataType=== 'increase') {
+        // If the graph is a percent-increase graph, we also want to show the absolute increase.
```

Comment:
Guha wants to see absolute increase on-hover for percent increase graphs.
So if it's a percent increase graph, we also have to calculate the difference.
Otherwise we can omit the extra calculation.